### DISCUSSION TITLE
Wallet connect not work on telegram mini app

### DISCUSSION THREAD

**Discussion by TrungHDinh** [WalletConnect/walletconnect-monorepo#4574] [Developer Support]
Hello, I am developing a Telegram mini app. It has a function to connect to a wallet. When it runs on a PC, everything works fine, but when it runs on a phone, it cannot connect to MetaMask.

I have tried the sample mini app from your repository at https://github.com/softstack/telegram-mini-app, but it encounters the same issue. 
When run on a phone it just stuck in here like this image.
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22341461/04b13fb5-9355-4642-aeed-178491fdcba9)

I try with android phone then it fire the error like this:
![image](https://github.com/WalletConnect/walletconnect-monorepo/assets/22341461/6a7339b7-06ed-4617-b38e-9a157b3649fa)

Anyone have same issue?

**Comment by GarageInc** [comment]
Had the same issue: https://github.com/DrKLO/Telegram/pull/1816

**Reply by AbdulRafaySiddiqui** [reply]
It's close so I guess no solution yet :(

**Comment by guotie** [comment]
How about iOS?

**Comment by jjohnson5253** [comment]
My workaround was just using the metamask sdk. Sorry WalletConnect. Now my telegram miniapp can connect a wallet via the metamask mobile app. https://github.com/MetaMask/metamask-sdk/blob/main/packages/examples/create-react-app/README.md

**Reply by quoctruongdev** [reply]
It's working fine. Thank you @jjohnson5253

**Reply by AjayiMike** [reply]
After setting up metamask for my telegram mini app, it works very well in the browser, but in telegram, I get this screen when I try to connect
![photo_2024-08-15_11-27-20](https://github.com/user-attachments/assets/077425a6-91f8-4bb1-921c-b68baa48cf45)

**Reply by AbdulRafaySiddiqui** [reply]
@jjohnson5253 are you connecting with the QR code? Cause directly connecting with deep link doesn't work, we just face the same issue.

**Reply by AjayiMike** [reply]
When on mobile, it immediately want to connect diretly to mobile metamask through deeplink. It only shows a QRCode if accessed through telegram desktop.

**Comment by Reglament989** [comment]
For anyone having similar problems, here's a dirty hack:
```js
window.open = (function (open) {
	return function (url, _, features) {
		return open.call(window, url, "_blank", features);
	};
})(window.open);
```

Quick reproduction [@wcminiappfixbot](https://t.me/wcminiappfixbot) [Source](https://github.com/Reglament989/wc-webview-reproduction)

IDK why, but this behavior has something to do with the target attribute, I think it's how webview works on android in general.

**Reply by 0xpaperhead** [reply]
Can you provide a little more context on your workaround? Did you just repurpose what window.open means?

**Reply by sivaram-undefy** [reply]
Thanks a ton !

**Reply by sp1r1don** [reply]
I can open the wallet now, but the wallet connection feature isn't working. It appears that WalletConnect won't function within an iframe. The issue seems to stem from the fact that the Telegram mini-app operates as a browser within an iframe.

**Reply by tx-tomcat** [reply]
any update on this?

**Reply by NOVALRR** [reply]
give me video tutorial step by step

**Reply by sp1r1don** [reply]
> give me video tutorial step by step

What?)

**Reply by qraxiss** [reply]
> For anyone having similar problems, here's a dirty hack:
> 
> ```js
> window.open = (function (open) {
> 	return function (url, _, features) {
> 		return open.call(window, url, "_blank", features);
> 	};
> })(window.open);
> ```
> 
> Quick reproduction [@wcminiappfixbot](https://t.me/wcminiappfixbot) [Source](https://github.com/Reglament989/wc-webview-reproduction)
> 
> IDK why, but this behavior has something to do with the target attribute, I think it's how webview works on android in general.

**this is working!**

**Comment by junkiet** [comment]
I dont think it is working, ask telegram to fix it.

**Comment by harissatti** [comment]
facing same issue .can't get the  exact information about this how we can set this ?

**Reply by sp1r1don** [reply]
There is no way to make it work. Only one solution I can imagine is some sort of proxy walletconnect socket using SSE.

**Reply by harissatti** [reply]
does this issue is from telegram side or walletconnect side?

**Reply by sp1r1don** [reply]
It's coming from Telegram. To be more precise miniApp injected with iframe and there is some restrictions, if I'm not mistaken

**Reply by sp1r1don** [reply]
As alternative you can try to find some others walletconnect like providers that using SSE

**Reply by harissatti** [reply]
did you have any link or info that get me their?

**Comment by creeping-vampires** [comment]
I had the same issue but you can use privy.io telegram auth for login within telegram mini app it will create smart account

**Comment by Nipunc55** [comment]
Had same issue, but now reown provide best solutions for connect wallet to telegram mini app

[https://docs.reown.com/appkit/features/telegram-mini-apps](url)
